comit/src/network/orderbook/order.rs

```diff
@@ -55,33 +84,94 @@ impl NewOrder {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct Order {
-    pub id: OrderId,
-    pub maker: PeerId,
-    pub position: Position,
-    pub bitcoin_amount: asset::Bitcoin,
-    pub bitcoin_ledger: ledger::Bitcoin,
-    pub bitcoin_absolute_expiry: u32,
-    pub ethereum_amount: asset::Erc20Quantity,
-    pub token_contract: identity::Ethereum,
-    pub ethereum_ledger: ledger::Ethereum,
-    pub ethereum_absolute_expiry: u32,
-}
-
 /// The position of the maker for this order. A BTC/DAI buy order,
 /// also described as an order that buys the trading pair BTC/DAI,
 /// means that the maker buys the base currency (in this case BTC) in
 /// return for DAI. A sell order means that the maker sells BTC and
 /// receives DAI.
-///
-/// Please note: we do not set the base currency to 1 and use rate
-/// (i.e., quote currency) and amount as is commonly done in Forex
-/// trading. We use the amounts of each currency to determine the
-/// rate.
 #[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
 #[serde(rename_all = "lowercase")]
 pub enum Position {
     Buy,
     Sell,
 }
+/// BTC to DAI conversion rate
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct BtcDaiRate(pub u64);
+
+impl Serialize for BtcDaiRate {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let precision = 10;
+        let base: u64 = 10;
+        let remainder = self.0 % base.pow(precision);
+        let integer = (self.0 - remainder) / base.pow(precision);
+        let string = format!("{}.{}", integer, remainder);
+
+        serializer.serialize_str(&string)
+    }
+}
+
+impl<'de> Deserialize<'de> for BtcDaiRate {
+    /// Decimal precision of 10 is selected because it is the largest possible
+    /// precision for BTC to DAI conversion rate.
+    /// Ethereum supports a precision of 18 and Bitcoin supports a precision of
+    /// 8 Therefore 10 is the maximum allowable precision for BTC to DAI
+    /// conversion
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let precision = 10;
+        let string = String::deserialize(deserializer)?;
+        let v: Vec<&str> = string.as_str().split('.').collect();
+
+        let integer = *v.first().unwrap();
+        let decimals = *v.last().unwrap();
```

Comment:
As noted in the other one, this can easily be triggered by user code and should be handled properly. You can instantiate a new error through `<D as Deserializer<'de>>::Error::custom` with a new error message.